zipline/finance/trading.py

```diff
@@ -204,17 +204,18 @@ def normalize_date(self, test_date):
         return pd.tseries.tools.normalize_date(test_date)
 
     def utc_dt_in_exchange(self, dt):
-        return pd.Timestamp(dt).tz_convert(self.exchange_tz)
+        return pd.Timestamp(dt, tz='UTC').tz_convert(self.exchange_tz)
 
     def exchange_dt_in_utc(self, dt):
         return pd.Timestamp(dt, tz=self.exchange_tz).tz_convert('UTC')
 
     def is_market_hours(self, test_date):
-        if not self.is_trading_day(test_date):
+        day = pd.Timestamp(self.utc_dt_in_exchange(test_date).date(), tz='UTC')
```

Comment:
what it's currently doing is taking open time and passing the open time to ```is_trading_day``` and ```get_open_and_close``` but open time can be on the previous when in UTC so you will be returned values for the previous day rather than the current day. I made a very similar PR last month #579.